game/views.py

```diff
@@ -152,27 +187,30 @@ def roll(self):
         if roll in self.currpity:
             self.currpity[roll] = 0
         return roll
-        
+
     def tickPity(self):
         for key in self.currpity:
             self.currpity[key] += 1
 
     def get_item(self, rarity):
-        item_names, item_chances = self.itemChanceLookup[rarity]["itemname"], self.itemChanceLookup[rarity]["chance"]
+        item_names, item_chances = self.itemChanceLookup[rarity][
+            "itemname"], self.itemChanceLookup[rarity]["chance"]
         roll = random.choices(item_names, item_chances)[0]
         return self.itemLookup[roll]
-    
+
     def get(self, request, game_id):
         if not self.populated:
             self.populate(game_id)
+        # if one of the data structures not populated, cannot roll.
+        if not self.raritylookup or not self.itemLookup:
+            return Response([])
         numrolls = int(request.GET.get('numrolls', ''))
         res = []
         for i in range(numrolls):
             rarity = self.roll()
             roll = RaritySerializer(self.raritylookup[rarity]).data
```

Comment:
rename to rarity?

Reply:
I think it's fine, I don't want to get it confused that it is the rarity itself, but the mapping of rarity id to rarity object

Reply:
call it rarityHash then. Roll is way more confusing lol